src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -593,8 +593,24 @@ public virtual Task<string> CopyNupkgAsync(string nupkgFilePath, CancellationTok
 
         public abstract Task<byte[]> GetArchiveHashAsync(HashAlgorithmName hashAlgorithm, CancellationToken token);
 
+        /// <summary>
+        /// Get contentHash for a signed package.
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns>null if the package is not signed</returns>
         public abstract string GetContentHashForSignedPackage(CancellationToken token);
```

Comment:
Do we even need this method now?  Can't we just call `GetContentHash(…)` everywhere?

Reply:
I think theorically we can merge both methods. The only thing is that [this](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs#L1173) method relies on null checking to fallback to the `SHA512` file we already have. By merging this two methods, we will be calculating the hash everytime instead of just reading it from the file. @jainaashish, what do you think?

Reply:
Pato you're right. Maybe we can refactor this code to take existing hash file path as well. Then first try to get content hash for signed package, else read from hash file if exists, else calculate the new hash.

And I'm fine to refactor it as part of separate PR.